pkg/webhook/resources/setting/validator.go

```diff
@@ -831,12 +835,20 @@ func (v *settingValidator) checkStorageNetworkValueVaild(setting *v1beta1.Settin
 	}
 
 	// check all VM are stopped, there is no VMI
-	vmis, err := v.vmis.List(metav1.NamespaceAll, labels.Everything())
+	vms, err := v.vmCache.List(metav1.NamespaceAll, labels.Everything())
 	if err != nil {
 		return err
 	}
-	if len(vmis) > 0 {
-		return werror.NewInvalidError("Please stop all VMs before configuring the storage-network setting", "value")
+
+	for _, vm := range vms {
+		isStopped, err := vmUtil.IsVMStopped(vm, v.vmiCache)
+		if err != nil {
+			return err
```

Comment:
perhaps also wrap with `werror.NewInternalError` or something makes more sense.

Reply:
Since [IsVMStopped()](https://github.com/harvester/harvester/blob/348cb16b6b9401f7de72aa6e92e848942f5fc8f4/pkg/util/virtualmachine/virtualmachine.go#L17-L53) reveals the actual error reason, I change as:
```
        isStopped, err := vmUtil.IsVMStopped(vm, v.vmiCache)
	if err != nil {
		return werror.NewInvalidError(err.Error(), "value")
	}
```
Thanks.